src/system.ts

```diff
@@ -68,14 +76,18 @@ export class System extends Element {
   protected formatter?: Formatter;
   protected startX?: number;
   protected lastY?: number;
-  protected parts: Required<SystemStave>[];
+  protected partStaves: Stave[];
+  protected partStaveInfos: StaveInfo[];
+  protected partVoices: Voice[];
   protected connector?: StaveConnector;
-  protected debugNoteMetricsYs?: { y: number; voice: Voice }[];
+  protected debugNoteMetricsYs?: { y: number; stave: Stave }[];
 
   constructor(params: SystemOptions = {}) {
     super();
     this.setOptions(params);
-    this.parts = [];
+    this.partStaves = [];
```

Comment:
Here too, not fully clear why these three are separate?

Reply:
See above